pkg/issuer/venafi/client/venaficlient.go

```diff
@@ -312,6 +318,48 @@ func httpClientForVcert(options *httpClientForVcertOptions) *http.Client {
 	}
 }
 
+// caBundleForVcertTPP is used to by ConnectionTrust and Client fields of vcert.Config.
+// This function sets appropriate CA based on provided bundle or kubernetes secret
+// If no custom CA bundle is configured, an empty byte slice is returned.
+// Assumes exactly one of the in-line/Secret CA bundles are defined.
+// If the `key` of the Secret CA bundle is not defined, its value defaults to
+// `ca.crt`.
+func caBundleForVcertTPP(tpp *cmapi.VenafiTPP, secretsLister internalinformers.SecretLister, namespace string) (caBundle []byte, err error) {
+	if len(tpp.CABundle) > 0 {
+		return tpp.CABundle, nil
+	}
+
+	secretRef := tpp.CABundleSecretRef
+	if secretRef == nil {
+		return nil, nil
+	}
+
+	var certBytes []byte
+	var ok bool
+
+	if secretRef != nil {
+		secret, err := secretsLister.Secrets(namespace).Get(secretRef.Name)
```

Comment:
What happens when the Secret does not exist and the controller has to wait for the Secret to be created?
Does the controller re-reconcile when the Secret is created? I think we might need some extra changes to make sure that works.

Reply:
Ah, nvm, I skipped over that change: [pkg/controller/issuers/checks.go](https://github.com/cert-manager/cert-manager/pull/7036/files#diff-4f757005715daaafbb177e36c76bffbc042f9bbf21dcd47aadc609b18ccd0c36R65-R70).